src/lib.rs

```diff
@@ -59,14 +61,28 @@ mod vbe_info;
 ///     println!("{:?}", boot_info);
 /// }
 /// ```
-pub unsafe fn load(address: usize) -> BootInformation {
-    assert_eq!(0, address & 0b111);
+pub unsafe fn load(address: usize) -> Result<BootInformation, MbiLoadError> {
+    let null_ptr = address == 0;
+    let eight_byte_aligned = 0 == address & 0b111;
+    if null_ptr || !eight_byte_aligned {
+        return Err(MbiLoadError::IllegalAddress);
+    }
     let multiboot = &*(address as *const BootInformationInner);
-    assert_eq!(0, multiboot.total_size & 0b111);
-    assert!(multiboot.has_valid_end_tag());
-    BootInformation {
-        inner: multiboot,
-        offset: 0,
+
+    // Check if total size is a multiple of 8.
+    // See MbiLoadError::IllegalTotalSize for comments
+    if 0 != multiboot.total_size & 0b111 {
+        return Err(MbiLoadError::IllegalTotalSize(multiboot.total_size));
+    }
+    if multiboot.has_valid_end_tag() {
```

Comment:
Might be cleaner to handle all of the invalid conditions with separate `if`s, and then leave the successful outcome as the default (i.e handle `!multiboot.has_valid_end_tag`, and then just return the `Ok` if you haven't returned yet).